pandas/tests/groupby/test_whitelist.py

```diff
@@ -164,33 +164,45 @@ def raw_frame():
 @pytest.mark.parametrize('axis', [0, 1])
 @pytest.mark.parametrize('skipna', [True, False])
 @pytest.mark.parametrize('sort', [True, False])
+@pytest.mark.parametrize('as_index', [True, False])
 def test_regression_whitelist_methods(
         raw_frame, op, level,
-        axis, skipna, sort):
+        axis, skipna, sort, as_index):
     # GH6944
     # GH 17537
     # explicitly test the whitelist methods
 
+    if not as_index and axis == 1:
+        pytest.skip('as_index=False only valid for axis=0')
+
     if axis == 0:
         frame = raw_frame
     else:
         frame = raw_frame.T
 
+    groupby_kwargs = {'level': level, 'axis': axis, 'sort': sort,
+        'as_index': as_index}
+    group_op_kwargs = {}
+    frame_op_kwargs = {'level': level, 'axis': axis}
     if op in AGG_FUNCTIONS_WITH_SKIPNA:
```

Comment:
I think this would be more clear with

```
# comment explaining why this is needed
if op in AGG_FUNCTION.S....:
     group_op_kwargs = .....
     frame_op_kwargs = ....
else:
    group......
    frame_op....
```

Reply:
So remove the idea of 'base arguments'? I thought it might be good to pull out the common arguments but maybe it's more readable the other way. The previous test code just seemed pretty repetitive to me.

Reply:
yeah just trying to make this simpler to read

Reply:
Sounds good, I will clean these tests up a bit and ping when everything is green!